google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -680,169 +637,18 @@ private T performDelete(
-              }
-            })
-        .toList();
-  }
-
-  List<SettableApiFuture<WriteResult>> getPendingFutures() {
-    return FluentIterable.from(pendingOperations)
-        .transform(
-            new Function<PendingOperation, SettableApiFuture<WriteResult>>() {
-              @Override
-              public SettableApiFuture<WriteResult> apply(PendingOperation input) {
-                return input.future;
-              }
-            })
-        .toList();
-  }
-
-  int getPendingOperationCount() {
-    return pendingOperations.size();
-  }
-
-  private ApiFuture<WriteResult> processOperation(DocumentReference documentReference) {
-    Preconditions.checkState(state == BatchState.OPEN, "Batch should be OPEN when adding writes");
-    SettableApiFuture<WriteResult> result = SettableApiFuture.create();
-    int nextBatchIndex = getPendingOperationCount();
-    pendingOperations.add(
-        new PendingOperation(nextBatchIndex, documentReference.getPath(), result));
-
-    if (getPendingOperationCount() == maxBatchSize) {
-      state = BatchState.READY_TO_SEND;
-    }
-
-    return result;
-  }
-
-  /**
-   * Resolves the individual operations in the batch with the results and removes the entry from the
-   * pendingOperations map if the result is not retryable.
-   */
-  void processResults(List<BatchWriteResult> results, boolean allowRetry) {
-    List<PendingOperation> newPendingOperations = new ArrayList<>();
-    for (int i = 0; i < results.size(); ++i) {
-      BatchWriteResult result = results.get(i);
-      PendingOperation operation = pendingOperations.get(i);
-
-      if (result.getException() == null) {
-        operation.future.set(new WriteResult(result.getWriteTime()));
-      } else if (!allowRetry || !shouldRetry(result.getException())) {
-        operation.future.setException(result.getException());
-      } else {
-        // Retry the operation if it has not been processed. Store the current index of
-        // pendingOperations to preserve the mapping of this operation's index in the underlying
-        // writes array.
-        newPendingOperations.add(new PendingOperation(i, operation.documentKey, operation.future));
-      }
-    }
-    pendingOperations = newPendingOperations;
-  }
-
-  private boolean shouldRetry(Exception exception) {
-    if (!(exception instanceof FirestoreException)) {
-      return false;
-    }
-    Set<Code> codes = FirestoreSettings.newBuilder().batchWriteSettings().getRetryableCodes();
-    Status status = ((FirestoreException) exception).getStatus();
-    for (Code code : codes) {
-      if (code.equals(Code.valueOf(status.getCode().name()))) {
-        return true;
-      }
-    }
-    return false;
+  List<WriteOperation> getWrites() {
+    return writes;
   }
 
-  void markReadyToSend() {
-    if (state == BatchState.OPEN) {
-      state = BatchState.READY_TO_SEND;
-    }
+  /** Get the number of writes. */
+  @VisibleForTesting
+  int getWriteCount() {
```

Comment:
We need to keep this method public for binary compatibility as I noted in the clirr file.

After making this public we no longer need the `@VisibleForTesting`.